sql3/sql_complex_test.go

```diff
@@ -127,6 +127,26 @@ func TestPlanner_SystemTableFanout(t *testing.T) {
 			t.Fatal(diff)
 		}
 	})
+
+	// additional testing for *ExecutionPlanner.mapper is going here because
+	// this is where the existing coverage comes from.
+	t.Run("MapperContextCancellation", func(t *testing.T) {
+		ctx, cancel := context.WithCancel(context.Background())
+		errch := make(chan error)
+		go func() {
+			_, _, _, err := sql_test.MustQueryRows(t, server, ctx, `select * from fb_performance_counters`)
```

Comment:
There is a subtle bug here, which we should probably fix, which is that we Can't Do This In A Goroutine.

The testing stuff strictly requires that, if you call `t.Fatalf` or the equivalent, you do so from the goroutine the test starts in.

In short: You should put the `cancel()` in a goroutine (after a sleep), and call MustQueryRows directly after launching that goroutine.